lib/internal/modules/esm/shared_constants.js

```diff
@@ -3,6 +3,40 @@
 
 'use strict';
 
+/*
+The shared memory area is divided in 4 32-bit long sections. It has to be 32-bit
+long as `Atomics.notify` only works with `Int32Array` objects.
+
+--32-bits-- --32-bits-- --32-bits-- --32-bits--
+    ^           ^           ^           ^
+    |           |           |           |
+    |           |           | NUMBER_OF_INCOMING_ASYNC_RESPONSES
+    |           |           |
+    |           | NUMBER_OF_INCOMING_MESSAGES
+    |           |
+    |   BOREDOM_OF_THREADS
+    |
+WORKER_TO_MAIN_THREAD_NOTIFICATION
+
+WORKER_TO_MAIN_THREAD_NOTIFICATION is only used to send notifications, its value is never used.
```

Comment:
This comment is fantastic, thank you for adding it.

```suggestion
WORKER_TO_MAIN_THREAD_NOTIFICATION is only used to send notifications. The value is never used;
the _change_ from one value to another is how one thread notifies the other.
```

Reply:
The change of value is not what's notify the other thread, that's `Atomics.notify`. I removed the `Atomics.store` since they were not used and confusing, and amended the comment, PTAL.